storage/replica.go

```diff
@@ -1484,9 +1485,21 @@ func (r *Replica) resolveIntents(ctx context.Context, intents []proto.Intent) {
 				if log.V(1) {
 					log.Warningc(ctx, "batch resolve failed: %s", err)
 				}
-				if _, ok := err.(*proto.RangeKeyMismatchError); !ok {
-					// TODO(tschottdorf)
-					panic(fmt.Sprintf("intent resolution failed, error: %s", err.Error()))
+				// At this point, as long as the local Replica accepts the
+				// request it should never fail. However, the replica may reject
+				// the request in certain cases (for example, if the replica has
+				// been removed from its range via a rebalancing a command).
+				// Therefore, we inspect the returned error to detect cases
+				// where the command was rejected, and can safely ignore those
+				// errors.
+				if err != multiraft.ErrGroupDeleted {
+					switch err.(type) {
+					case *proto.RangeKeyMismatchError:
+					case *proto.NotLeaderError:
+					default:
+						// TODO(tschottdorf): Does this need to be a panic?
+						panic(fmt.Sprintf("intent resolution failed with unexpected error: %s", err.Error()))
```

Comment:
in this instance, i think it's just superfluous. in other instances, it might (a) allocate an extra string and (b) prevent the use of a `String` method, if one is present.